pkg/controller/service/servicecontroller.go

```diff
@@ -125,16 +127,33 @@ func New(
 		nodeLister:       nodeInformer.Lister(),
 		nodeListerSynced: nodeInformer.Informer().HasSynced,
 		workingQueue:     workqueue.NewNamedDelayingQueue("service"),
+		endpoints: make(map[string]int),
 	}
 
 	serviceInformer.Informer().AddEventHandlerWithResyncPeriod(
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: s.enqueueService,
 			UpdateFunc: func(old, cur interface{}) {
+				endpointsChanged := false
 				oldSvc, ok1 := old.(*v1.Service)
 				curSvc, ok2 := cur.(*v1.Service)
+				oldEndpoints := s.endpoints[oldSvc.ObjectMeta.Name]
+				newEndpoints := s.getNumberOfEndpoints(oldSvc)
```

Comment:
@MrHohn We need this implementation to handle updates for services of type=loadbalancer. And that's a choice that each Cloud Provider would have to make (that should be made configurable), if it uses 'nodeIPs' versus 'podIPs' as the LB members for such services.

I think it is reasonable to implement a check to decide whether the implementation uses NodeIp versus PodIp, and design the corse of this implementation.